src/main/java/org/spdx/sbom/gradle/project/SpdxPackageInfo.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 The Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spdx.sbom.gradle.project;
+
+import java.util.Optional;
+import org.immutables.serial.Serial;
+import org.immutables.value.Value.Immutable;
+import org.spdx.sbom.gradle.SpdxPackageExtension;
+
+@Immutable
+@Serial.Version(1)
+public interface SpdxPackageInfo {
+
+  String getName();
+
+  String getVersion();
+
+  boolean getCreatePackage();
+
+  Optional<ScmInfo> getScmInfo();
+
+  Optional<String> getSupplier();
+
+  static SpdxPackageInfo from(SpdxPackageExtension ext) {
+    if (ext == null) {
+      throw new NullPointerException("Extension value cannot be null");
+    }
+    var builder =
+        ImmutableSpdxPackageInfo.builder()
+            .name(ext.getName().get())
+            .version(ext.getVersion().get())
+            .createPackage(ext.getCreatePackage().get())
+            .supplier(Optional.ofNullable(ext.getSupplier().getOrNull()));
+
+    var scm = ext.getScm();
+    if (scm.getRevision().isPresent() && scm.getTool().isPresent() && scm.getUri().isPresent()) {
+      builder.scmInfo(
+          ScmInfo.from(scm.getTool().get(), scm.getUri().get(), scm.getRevision().get()));
+    }
+    if ((scm.getUri().isPresent() && !scm.getRevision().isPresent())
+        || (!scm.getUri().isPresent() && scm.getRevision().isPresent())) {
```

Comment:
Hmm I'm not sure what happened to my first copy of this comment but I don't see it anymore.

I think if `tool` is not present,  line 50 will be skipped but this error won't trigger. Do we want to check for `tool` here? Do we want to skip checking for `tool` on line 49?

Reply:
Tool defaults to git. But that's obviously not apparent here. Maybe I should make it more apparent

Reply:
Yeah I noticed that `tool` defaults to "git" - I'm just not sure why we would check for it in only one place (line 49) and not the other (line 53-54)

Reply:
I added a check to make it explicit.

Reply:
ok cool. I think we also wouldn't need the `scm.getTool().isPresent()` on line 49 if we moved this check to be earlier.

Still looks good to me